src/imitation/scripts/parallel.py

```diff
@@ -47,15 +49,28 @@ def parallel(
       local_dir: `local_dir` argument to `ray.tune.run()`.
       upload_dir: `upload_dir` argument to `ray.tune.run()`.
     """
+    # Basic validation for config options before we enter parallel jobs.
+    for name in base_named_configs:
+        assert isinstance(name, str)
+    for k in base_config_updates:
+        assert isinstance(k, str)
+    assert isinstance(search_space["named_configs"], collections.abc.Sequence)
+    assert isinstance(search_space["config_updates"], collections.abc.Mapping)
+
     # Explicitly set `data_dir` if parallelizing `train_adversarial`.
-    # We need this to automatically find rollout pickles Ray will automatically
-    # change the working directory for each Raylet.
+    # We need this to automatically find rollout pickles because Ray
+    # sets a new working directory for each Raylet.
     if sacred_ex_name == "train_adversarial":
         if "data_dir" not in base_config_updates:
-            base_config_updates["data_dir"] = os.path.join(os.getcwd(), "data/")
+            data_dir = os.path.join(os.getcwd(), "data/")
+            base_config_update = dict(base_config_updates)
+            base_config_update["data_dir"] = data_dir
 
     trainable = _ray_tune_sacred_wrapper(
-        sacred_ex_name, run_name, base_named_configs, base_config_updates
+        sacred_ex_name,
+        run_name,
+        copy.deepcopy(base_named_configs),
```

Comment:
Aren't `base_named_configs` just a list of strings? Can we just do `list(base_named_configs)`? I'm not sure I even understand why this is needed -- is `base_named_configs` a `FrozenList`? Good to add a comment explaining this.

Reply:
I added in the `deepcopy` to check if non-forked Sacred would work, and left it in for a confused reason that probably isn't worth elaborating on.

non-forked Sacred didn't work because it would deterministically fail on the issue fixed by https://github.com/IDSIA/sacred/pull/740.